core/services/chainlink/config.go

```diff
@@ -67,6 +74,60 @@ func (c *Config) SetDefaults() {
 	//TODO terra and solana defaults https://app.shortcut.com/chainlinklabs/story/37975/chains-nodes-should-be-read-from-the-config-interface
 }
 
+type Secrets struct {
+	config.Secrets
+}
+
+func (s *Secrets) Validate() error {
+	return config.Validate(s)
+}
+
+// Use ENV vars or flags to override secrets
+func (s *Secrets) SetOverrides(context *cli.Context) error {
+	// Override DB and Explorer secrets from ENV vars, if present
```

Comment:
ENV var names here duplicate schema.go but I believe schema.go will get deprecated at some point, right ... ?

Reply:
Yeah, shouldn't have to depend on any legacy logic ideally. Would be ok to share some code in the mean time if it's clean (e.g. package envvar)